src/ImageSharp/Common/Helpers/RuntimeUtility.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Six Labors.
+// Licensed under the Six Labors Split License.
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace SixLabors.ImageSharp.Common.Helpers
+{
+    /// <summary>
+    /// A helper class that with utility methods for dealing with references, and other low-level details.
+    /// </summary>
+    internal static class RuntimeUtility
+    {
+        /// <summary>
+        /// Swaps the two references.
+        /// </summary>
+        /// <typeparam name="T">The type to swap.</typeparam>
+        /// <param name="a">The first item.</param>
+        /// <param name="b">The second item.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void Swap<T>(ref T a, ref T b)
+        {
+            // Tuple swap uses 2 more IL bytes
+#pragma warning disable IDE0180 // Use tuple to swap values
```

Comment:
Maybe do this on file level instead of the two spots where this is happening?

Reply:
Good shout